Server.py

```diff
@@ -0,0 +1,224 @@
+import errno
+import select
+import socket
+import traceback
+
+import base
+import Client
+import CustomExceptions
+import util
+
+CLOSE, SERVER, CLIENT = range(3)
+CRLF = "\r\n"
+END_HEARDER = 2 * CRLF
+
+
+class Server(base.Base):
+    _database = {}
+
+    def __init__(
+        self,
+        buff_size,
+        base_directory
+    ):
+        super(Server, self).__init__()
+        self._buff_size = buff_size
+        self._run = True
+        self.logger.info("Initialized server, buff size '%d'", buff_size)
+        self._base_directory = base_directory
+
+    def terminate(self):
+        self._run = False
+
+    def add_server(
```

Comment:
We don't want to recive more information if we want to close the socket.
But I changed everything else